### DISCUSSION TITLE
Position Updates of Multiple Nodes on Drag Stop

### DISCUSSION THREAD

**Discussion by abbieschenk** [xyflow/xyflow#1240] [Q&A]
Hello,

I need to save / persist the position of Nodes with an API call after they are moved around. This works for one node by using "onNodeDragStop(event, node)."

It doesn't work when selecting multiple nodes with Cmd and then dragging them all around, as the onNodeDragStop event is only called on the first selected node — note I don't mean shift + dragging around the nodes and moving them that way, as I can get that working fine.

I tried to fix this by using `const selectedElements = useStoreState(actions => actions.selectedElements);` as per Issue #492. But these elements do not have an updated position, as it appears they only get initialized once and then not updated.

I was wondering if there is a way to get all the updated positions of all nodes that have moved on a drag event. Thank you!

**Comment by FredrikAugust** [comment]
Bump! Any updates on this?

**Comment by SanditZZ** [comment]
Hi, 
My solution is to use the store from `useStoreApi()` together with `onNodeDragStop` and get the selected nodes (node.selected) from the state.nodes, then you can do whatever you need with the selected nodes. Would like to hear if anyone has a different solution!

Example:
```ts
import {
  useStoreApi,
} from "@xyflow/react";
```

```ts
  const store = useStoreApi();

  const onNodeDragStop = useCallback(
    (_: React.MouseEvent, node: Node) => {
      // Get selected nodes from store
      const state = store.getState();
      const selectedNodeIds = state.nodes.filter((n: any) => n.selected).map((n: any) => n.id);

      console.log("selectedNodes", selectedNodeIds);
    },
    [store]
  );
```

**Comment by ivhmao** [comment]
Hi,
Today OnNodeDrag type has an additional _nodes_ property: 
```
export type OnNodeDrag<NodeType extends Node = Node> = (
  event: ReactMouseEvent,
  node: NodeType,
  nodes: NodeType[]
) => void;

```

So I guess it should look something like this:
```
  const onNodeDragStop: OnNodeDrag<MyNode> = useCallback(
    async (event, _, nodes) => {
      console.log("selectedNodes", nodes);
      for (const node of nodes){
        try {
          const updatedNode = {
            ...node.data,
            Position: {
              x: node.position.x,
              y: node.position.y
            }
          };

          await updateNodeInAPI(updatedNode).unwrap();

          console.log(`Position of "${node.id}" saved`);

        } catch (err) {
          console.error(err);
          console.log( `Error saving position for "${node.id}");
        }
      }
    },[updateNodeInAPI]); 
```
